ts/handler_utils/caller.py

```diff
@@ -0,0 +1,16 @@
+from abc import ABC
+
+
+class InitCaller(ABC):
+    def __call__(self, *args):
+        if self._prev:
+            self._prev(*args)
+
+        return self._method(*args)
+
+
+class PipeCaller(ABC):
```

Comment:
wdyt about this just being a python iterable instead?

Reply:
Okay, I think now I see what the confusion is. Will probably redefine this interface to not use interitance and put the constructor in the Caller. The intention is this:

```python
def foo(a):
   return a+1
def bar(b):
  return b+2
pp = PipeCaller(foo)
pp = PipeCaller(pp, bar)
c=pp(1) # c==4
```